common/scripts/mesh-helper.sh

```diff
@@ -43,21 +43,34 @@ EOF
   fi
 }
 
-generate_br_lan_ip() {
+generate_bridge_ip() {
   local mesh_if_mac
-  mesh_if_mac="$(ip -brief link | grep "$MESH_VIF" | awk '{print $3; exit}')"
+
+  bridge_name=$(echo "$BRIDGE" | cut -d' ' -f1)
+
+  mesh_if_mac="$(ip -brief link | grep "${id0_MESH_VIF}" | awk '{print $3; exit}')"
+  if [ -z "$mesh_if_mac" ]; then
+      echo "generate_bridge_ip: MAC not found for id0_MESH_VIF! Configuration error?" > /dev/kmsg
+      mesh_if_mac="$(ip -brief link | grep "eth0" | awk '{print $3; exit}')"
+  fi
   local ip_random
   ip_random="$(echo "$mesh_if_mac" | cut -b 16-17)"
-  br_lan_ip="192.168.1."$((16#$ip_random))
+  bridge_ip="192.168.1."$((16#$ip_random))
+
+  # legacy support
+  br_lan_ip=$bridge_ip
 }
 
 source_configuration() {
-  if [ -f "/opt/mesh.conf" ]; then
+  # $1: index of the mesh.conf file to be used
+
+  if [ -f "/opt/${1}_mesh.conf" ]; then
     # Calculate hash for mesh.conf file
```

Comment:
Please explain why, I for one struggle to understand what this is about :)

If this is a way to help parse the configuration then maybe we need to brainstorm a better way such as breaking it into a conf.d/ directory for example as it more and more common practice.

Otherwise checking if a file changed is already doable better with `diff` already (e.g. a hash will be completely different if only a newline was added for example, while with `diff` such insignificant changes can be ignored) or if we really want to detect any sort of byte difference then these steps can be replaced simply with the `cmp` command :)

Reply:
Some pseudo lines added.  Maybe that helps.   Hash is for to detect modified file without using correct interfaces.
```
source_configuration() {
  # $1: radio index of the mesh.conf file to be used

  # if we have radio specific config written into device over NATS
  if [ -f "/opt/${1}_mesh.conf" ]; then
    if hash ok; then
      source /opt/"${1}"_mesh.conf
    else
      # use default mesh, if hash is not ok
      if default available (generated in first boot); then
        source /opt/mesh_default.conf
      else
        echo "no valid mesh configuration found!"
		exit 1
      fi
    fi
  elif if we don't have radio specific file written --> use default; then
      source /opt/mesh_default.conf
  else
      echo "no valid mesh configuration found!"
      exit 1
  fi
}
```